src/coreclr/src/jit/hwintrinsic.cpp

```diff
@@ -789,6 +789,13 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
                         // to the code generator. May encode the overload info in other way.
                         retNode->AsHWIntrinsic()->gtSIMDBaseType = JITtype2varType(corType);
                     }
+#ifdef TARGET_ARM64
+                if ((intrinsic == NI_AdvSimd_AddWideningUpper) || (intrinsic == NI_AdvSimd_SubtractWideningUpper))
+                {
+                    assert(varTypeIsSIMD(op1->TypeGet()));
+                    retNode->AsHWIntrinsic()->SetOtherBaseType(getBaseTypeOfSIMDType(argClass));
```

Comment:
Why do widening and narrowing need an "other" base type? The return type is always twice the size and of the same signedness of the inputs for these operations.

Reply:
> Why do widening and narrowing need an "other" base type?

Narrowing intrinsics do not need - only AddWideningUpper and SubractWideningUpper do.

They don't need the operand 1 type itself (which is always going to be TYP_SIMD16) but the operand 1 element type so I can distinguish between each pair of these three
```c#
Vector128<int> AddWideningUpper(Vector128<short> left, Vector128<short> right);
Vector128<short> AddWideningUpper(Vector128<short> left, Vector128<sbyte> right);
Vector128<int> AddWideningUpper(Vector128<int> left, Vector128<short> right);
```

Reply:
Why can't we just use the second argument for the base type? That is what `HW_Flag_BaseTypeFromSecondArg` is for.

Reply:
Oh nevermind, I missed that `int = short + short` and `int = int + short` need to be differentiated

Reply:
Interesting - I renamed the `gtIndexBaseType` to `gtOtherBaseType` when I restructured the IR, because I figured there would be cases aside from gather where we'd need an additional base type - I didn't think it would be used that quickly :-)